### DISCUSSION TITLE
Delete image

### DISCUSSION THREAD

**Discussion by Fredxd** [ueberdosis/tiptap#3017] [General Questions] [answered]
Hi,

I want to delete an image node added with extension-image, but i don't know how do it.
I have try to set **src** attribute to **null**, but img tag is not cleared.
I have try also to use ` editor.commands.deleteNode('image')` but nothing happens.

Is there a way to do this, either via the TipTap API or via the Prosemirror node ?
If not, can you add a **removeImage** method in extension or can propose a PR ?

Thx in advance

**Comment by Fredxd** [comment]
Hi,
Any idea about this?

Regards

**Comment by AbhayVAshokan** [comment, marked as answer]
I had a use case for deleting the selected image. I used `editor.commands.deleteSelection()` for the same.

**Reply by Fredxd** [reply]
It's work fine. Thanks a lot

**Reply by Ayomikuolatunji** [reply]
After deleting the image, it does not allow me to be able to add the same image again. Any good solution ?

**Reply by Hugo-Lorenzoni** [reply]
same problem than @Ayomikuolatunji  here...

**Reply by Hugo-Lorenzoni** [reply]
why is the unsetImage() function missing tho ?

**Reply by Hugo-Lorenzoni** [reply]
the reason why I wasn't able to add the same image again came from the fact that the "onChange" event of my input field wasn't firing because the image didn't change !
so fix it you need a ref to your input and reset the value each time you handle the setImage : 
```
if (inputRef.current) {
   inputRef.current.value = "";
}
```
nothing to do with the tiptap extension (but I still think that having a method like unsetImage() could help)
